tutorials/tour/_posts/2017-02-13-inner-classes.md

```diff
@@ -31,50 +31,32 @@ class Graph {
   }
 }
 ```
- 
-In our program, graphs are represented by a list of nodes. Nodes are objects of inner class `Node`. Each node has a list of neighbours, which get stored in the list `connectedNodes`. Now we can set up a graph with some nodes and connect the nodes incrementally:
- 
-```tut
-object GraphTest extends App {
-  val g = new Graph
-  val n1 = g.newNode
-  val n2 = g.newNode
-  val n3 = g.newNode
-  n1.connectTo(n2)
-  n3.connectTo(n1)
-}
-```
- 
-We now enrich the above example with types to state explicitly what the type of the various defined entities is:
- 
+This program represents a graph as a list of nodes (`var nodes`). Each node knows which other nodes it's connected to (`connectedNodes`). The `class Node` is a _path-dependent type_ because it is nested in the `class Graph`. Therefore, everything in the `connectedNodes` must be created using the class attached to Graph.
+
 ```tut
-object GraphTest extends App {
-  val g: Graph = new Graph
-  val n1: g.Node = g.newNode
-  val n2: g.Node = g.newNode
-  val n3: g.Node = g.newNode
-  n1.connectTo(n2)
-  n3.connectTo(n1)
-}
+val graph1: Graph = new Graph
+val node1: graph1.Node = graph1.newNode
+val node2: graph1.Node = graph1.newNode
+val node3: graph1.Node = graph1.newNode
+node1.connectTo(node2)
+node3.connectTo(node1)
 ```
- 
-This code clearly shows that a node type is prefixed with its outer instance (which is object `g` in our example). If we now have two graphs, the type system of Scala does not allow us to mix nodes defined within one graph with the nodes of another graph, since the nodes of the other graph have a different type.
+Notice how the type of all of the nodes is `graph1.Node`. This is because when we call `graph1.newNode` which calls `new Node`, the method is using the instance of `Node` specific to the instance `graph1`.
```

Comment:
After the "Notice how" sentence, perhaps insert: "We have explicitly supplied that type, but the compiler would infer it if we had left it out."